### ISSUE TITLE
Building gwt generated project with setup-tool gets many deprecation warnings.

### ISSUE THREAD

**Issue by patton73** [libgdx/libgdx#7123]
I used latest build-tool downloaded from the site.
Compilation succeeded. But a lot of warnings were printed.

Compiling module it.rebirthproject.testgwt.GdxDefinition
   Adding '1' new generated units
      [WARN] Warnings in 'generated://F298620D7D37C41F444E1CD88D4BC47E/com/badlogic/gwtref/client/IReflectionCacheGenerated.java'
         [WARN] Line 10986: Referencing field 'com.badlogic.gdx.graphics.g2d.SpriteBatch.defaultVertexDataType': field 'com.badlogic.gdx.graphics.g2d.SpriteBatch.defaultVertexDataType' is deprecated
         [WARN] Line 10987: Referencing field 'com.badlogic.gdx.graphics.g2d.SpriteBatch.defaultVertexDataType': field 'com.badlogic.gdx.graphics.g2d.SpriteBatch.defaultVertexDataType' is deprecated
         [WARN] Line 12610: Referencing field 'com.badlogic.gdx.math.Matrix4.tmp': field 'com.badlogic.gdx.math.Matrix4.tmp' is deprecated
         [WARN] Line 15057: Referencing method 'com.badlogic.gdx.math.Vector2.angle': method 'com.badlogic.gdx.math.Vector2.angle()' is deprecated
         [WARN] Line 15058: Referencing method 'com.badlogic.gdx.math.Vector2.angle': method 'com.badlogic.gdx.math.Vector2.angle(Lcom/badlogic/gdx/math/Vector2;)' is deprecated
         [WARN] Line 15063: Referencing method 'com.badlogic.gdx.math.Vector2.setAngle': method 'com.badlogic.gdx.math.Vector2.setAngle(F)' is deprecated
         [WARN] Line 15066: Referencing method 'com.badlogic.gdx.math.Vector2.rotate': method 'com.badlogic.gdx.math.Vector2.rotate(F)' is deprecated
         [WARN] Line 15067: Referencing method 'com.badlogic.gdx.math.Vector2.rotateAround': method 'com.badlogic.gdx.math.Vector2.rotateAround(Lcom/badlogic/gdx/math/Vector2;F)' is deprecated
         [WARN] Line 15329: Referencing method 'com.badlogic.gdx.scenes.scene2d.Actor.ancestorsVisible': method 'com.badlogic.gdx.scenes.scene2d.Actor.ancestorsVisible()' is deprecated
   Adding '1' new generated units
      [WARN] Warnings in 'generated://7D2CBFAAD6BEB73A4BB1566947BD8DAE/com/badlogic/gwtref/client/IReflectionCache2Generated.java'
         [WARN] Line 14838: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActor': method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActor(Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 14839: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorAt': method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorAt(ILcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 14840: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorBefore': method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorBefore(Lcom/badlogic/gdx/scenes/scene2d/Actor;Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 14841: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorAfter': method 'com.badlogic.gdx.scenes.scene2d.ui.Container.addActorAfter(Lcom/badlogic/gdx/scenes/scene2d/Actor;Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15229: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.setWidget': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.setWidget(Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15230: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.getWidget': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.getWidget()' is deprecated
         [WARN] Line 15231: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActor': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActor(Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15232: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorAt': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorAt(ILcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15233: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorBefore': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorBefore(Lcom/badlogic/gdx/scenes/scene2d/Actor;Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15234: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorAfter': method 'com.badlogic.gdx.scenes.scene2d.ui.ScrollPane.addActorAfter(Lcom/badlogic/gdx/scenes/scene2d/Actor;Lcom/badlogic/gdx/scenes/scene2d/Actor;)' is deprecated
         [WARN] Line 15339: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.SelectBox.showList': method 'com.badlogic.gdx.scenes.scene2d.ui.SelectBox.showList()' is deprecated
         [WARN] Line 15341: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.SelectBox.hideList': method 'com.badlogic.gdx.scenes.scene2d.ui.SelectBox.hideList()' is deprecated
         [WARN] Line 15802: Referencing method 'com.badlogic.gdx.scenes.scene2d.ui.Tree.getNodes': method 'com.badlogic.gdx.scenes.scene2d.ui.Tree.getNodes()' is deprecated
         [WARN] Line 16204: Referencing method 'com.badlogic.gdx.scenes.scene2d.utils.Selection.hasItems': method 'com.badlogic.gdx.scenes.scene2d.utils.Selection.hasItems()' is deprecated
         [WARN] Line 16549: Referencing method 'com.badlogic.gdx.utils.JsonValue.size': method 'com.badlogic.gdx.utils.JsonValue.size()' is deprecated

**Comment by tommyettinger** [comment]
Those warnings (note, **they are not full-blown errors**, and do not impact your build in any way) come from how the reflection cache is built, which has to compile, or traverse in some way, all the methods in classes it can reflect upon. Referencing a deprecated method in an otherwise non-deprecated class emits that warning that a method exists and is deprecated, even if it isn't used by your code or called by libGDX. GWT normally doesn't even have a reflection cache, and lacks any way of handling reflection out-of-the-box without libGDX, so vanilla GWT projects don't encounter this type of warning unless they actually are using deprecated code. In a libGDX project, the warnings show up even if nothing uses the deprecated code.

So in other words, it doesn't affect you or your build, and the only way to get rid of it fully would be to remove all deprecated declarations, or to remove the `@Deprecated` annotation all over (which is the same as no longer deprecating any code). The first would break anything that used an older version of libGDX and the deprecated code, even if it wasn't deprecated in the older version. Since backwards-compatibility is a priority, we aren't doing that. The second isn't a good idea either, things are deprecated to discourage their use in favor of a different way of doing the same thing or accomplishing a similar thing.

**Comment by patton73** [comment]
So basically you deprecate code but never remove it for backward compatibility? So what's the meaning of code deprecation? 
However since this not affect my build, this is not a big problem. But a major version should remove that deprecated code.
Thanks.

**Comment by tommyettinger** [comment]
In many cases the point is to discourage the use in new code, while permitting older code to still function. There's a lot of old libGDX codebases out there, and mostly they aren't that difficult to bring up to date because we have pretty good backwards compatibility. To contrast, the Android APIs are a hellscape of features added, quickly deprecated in favor of a newer (not necessarily any better) solution, and removed for newer devices but not older ones, some of which may not have the new solution available. Just look at file storage on Android, ugh...
